monitor/src/main/java/edu/asu/diging/monitor/web/admin/ModifyAppController.java

```diff
@@ -0,0 +1,49 @@
+package edu.asu.diging.monitor.web.admin;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import edu.asu.diging.monitor.core.model.IApp;
+import edu.asu.diging.monitor.core.service.IAppManager;
+import edu.asu.diging.monitor.web.admin.forms.AppForm;
+
+@Controller
+public class ModifyAppController {
+	
+	
+	@Autowired
+	private IAppManager appManager;
+
+	@RequestMapping(value="/admin/apps/{id}/modify", method= RequestMethod.GET)
+	public String show(Model model, @PathVariable("id") String id) {
+		model.addAttribute("appForm", appManager.getApp(id));
+		return "admin/apps/show";
```

Comment:
hm, so this works but it's pretty error prone. It works because the App class and AppForm class have the same properties. If, however, either class is changed then you will have site effects. E.g. if one of properties is renamed, the compiler wouldn't catch it most likely but you would run into issues when you try to display it. I'm fine with leaving this the way it is for now (since we're not very likely to change this app a lot), but please make a comment in the code here.

Reply:
?